Sources/ForageSDK/Foundation/ForageSDK+ForageServices.swift

```diff
@@ -94,7 +94,16 @@ extension ForageSDK: ForageSDKService {
                 completion(.failure(CommonErrors.INCOMPLETE_PIN_ERROR))
                 return
             }
-                        
+            
+            // This block is used for tracking important Metrics!
+            // -----------------------------------------------------
+            let responseMonitor = CustomerPerceivedResponseMonitor.newMeasurement(
+                vaultType: foragePinTextField.collector!.getVaultType(),
```

Comment:
Any way to avoid this unwrap or are we just gonna roll the dice?

Reply:
We used to force unwrap `collector` in this other part of the same method – so the force-wrapping from my diff does not make this method any more prone to crashes than it already was.

However, i think we can refactor the force unwrapping and handle this more gracefully; I'll handle this as part of this cleanup ticket that [is on my plate for this week](https://linear.app/joinforage/issue/FX-631/ios-clean-up-checkbalance-and-capturepayment)